plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/AppEngineContentProvider.java

```diff
@@ -164,6 +243,9 @@ public boolean hasChildren(Object element) {
       AppEngineStandardProjectElement projectElement = (AppEngineStandardProjectElement) element;
       return projectElement.getConfigurations() != null
           && projectElement.getConfigurations().length > 0;
+    } else if (element instanceof AppEngineResourceElement) {
+      // none of our descriptor models have children
+      return false;
     }
     IProject project = getProject(element);
     if (project == null) {
```

Comment:
At the end of this method is

`return webProject != null ? webProject.getConfigurations().length > 0 : true;`

. Now I wonder if it should return false when `webProject == null`.

Reply:
If `webProject == null` then we haven't actually loaded information about the project, so we return `true` to err on the side of caution.  (This result is used by the `Tree` to decide whether to have a expand-to-show-children triangle: if it returns false, then it will never show the expand-triangle, even if one should be there.)

Unfortunately the use of a LoadingCache, which can't have null values, means that we can't differentiate between non-App Engine projects and projects we haven't checked yet.